### PR TITLE
AMLS 5087 | Payments Auth update

### PR DESCRIPTION
**Pull request by owen-opencast**
Updated Payments Auth

## How Has This Been Tested?

Spec tests and coverage report (Statement coverage 90.28% | Branch coverage....: 86.28%) 

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [ ] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: app/controllers/payments/TypeOfBankController.scala [hunk 1]
```diff
@@ -16,57 +16,55 @@
...
               Redirect(controllers.payments.routes.BankDetailsController.get(data.isUK).url)
             }
 
           case f: InvalidForm => Future.successful(BadRequest(views.html.payments.type_of_bank(f)))
         }
   }
 
-  private def doAudit(ukBank: Boolean)(implicit hc: HeaderCarrier, ac: AuthContext) = {
+  private def doAudit(ukBank: Boolean, amlsRefNumber: Option[String], accountTypeId: (String, String))(implicit hc: HeaderCarrier) = {
+    val x = feeResponseService.getFeeResponse(amlsRefNumber.get, accountTypeId)
```
[the remark this answers is not in the document]
Comment (ajchilts) [reply app/controllers/payments/TypeOfBankController.scala]: If we need the AMLS ref no here and we can't proceed without it; we should do a get or else and raise if its not found?

File: app/controllers/payments/WaysToPayController.scala [hunk 2]
```diff
@@ -83,12 +82,12 @@ class WaysToPayController @Inject()(
...
   def progressToPayment(fn: (FeeResponse, String, String) => Future[Result])
                        (errorMessage: String)
-                       (implicit ac: AuthContext, hc: HeaderCarrier): Future[Result] = {
+                       (implicit hc: HeaderCarrier, request:AuthorisedRequest[_]): Future[Result] = {
 
     val submissionDetails = for {
-      amlsRefNo <- OptionT(authEnrolmentsService.amlsRegistrationNumber)
-      (_, detailedStatus) <- OptionT.liftF(statusService.getDetailedStatus(amlsRefNo))
-      fees <- OptionT(feeResponseService.getFeeResponse(amlsRefNo))
+      amlsRefNo <- OptionT(authEnrolmentsService.amlsRegistrationNumber(request.amlsRefNumber, request.groupIdentifier))
```
Comment (ajchilts) [inline app/controllers/payments/WaysToPayController.scala]: As above; we should not use this method. Get it from the authorised request which retrieves the enrolments up front.

File: test/controllers/payments/WaysToPayControllerSpec.scala [hunk 3]
```diff
@@ -155,12 +157,17 @@ class WaysToPayControllerSpec extends AmlsSpec with AmlsReferenceNumberGenerator
               "waysToPay" -> WaysToPay.Card.entryName
             )
 
+//            when {
+//              controller.paymentsService.requestPaymentsUrl(any(), any(), any(), any(), any())(any(), any(), any())
+//            } thenReturn Future.successful(NextUrl("/payments-next-url"))
+
```
Comment (ajchilts) [inline test/controllers/payments/WaysToPayControllerSpec.scala]: Can we please remove this if not needed? Should this test be there and should it pass? I.e. did it pass before the auth changes?

File: app/controllers/payments/TypeOfBankController.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (owen-opencast) [reply app/controllers/payments/TypeOfBankController.scala]: Ah, well spotted... I'll remove this line, I was using it to test the feeResponseService while debugging

File: app/controllers/payments/BankDetailsController.scala [hunk 4]
```diff
@@ -19,31 +19,30 @@ package controllers.payments
...
-  def get(isUK: Boolean = true) = Authorised.async {
-    implicit authContext =>
+  def get(isUK: Boolean = true) = authAction.async {
       implicit request =>
         (for {
-          submissionRequestStatus <- OptionT.liftF(dataCacheConnector.fetch[SubmissionRequestStatus](SubmissionRequestStatus.key))
-          status <- OptionT.liftF(statusService.getStatus)
-          amlsRegistrationNumber <- OptionT(authEnrolmentsService.amlsRegistrationNumber)
-          fees <- OptionT(feeResponseService.getFeeResponse(amlsRegistrationNumber))
+          amlsRefNo <- OptionT(authEnrolmentsService.amlsRegistrationNumber(request.amlsRefNumber, request.groupIdentifier))
```
[the remark this answers is not in the document]
Comment (owen-opencast) [reply app/controllers/payments/BankDetailsController.scala]: Using the updated version of authEnrolmentsService.amlsRegistrationNumber by @slavvco, as it is being mocked in tests

File: app/controllers/payments/WaysToPayController.scala (hunk 2 above)
Comment (owen-opencast) [reply app/controllers/payments/WaysToPayController.scala]: Using the updated version of authEnrolmentsService.amlsRegistrationNumber by @slavvco, as it is being mocked in tests

File: test/controllers/payments/WaysToPayControllerSpec.scala (hunk 3 above)
Comment (owen-opencast) [reply test/controllers/payments/WaysToPayControllerSpec.scala]: Removed

File: app/controllers/payments/TypeOfBankController.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (owen-opencast) [reply app/controllers/payments/TypeOfBankController.scala]: Removed, along with the println() below it

**Comment by owen-opencast** [comment]
I've removed the commented out code, and variable I'd added for debugging. authEnrolmentsService.amlsRegistrationNumber kept, as discussed.

**Pull request merged**